src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build();
+        this.schedulePass();
+      } else {
+        this.pendingBuildResources_.push(element);
+        this.buildReadyResources_();
+      }
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
```

Comment:
Yes that assumption is correct, but not sure if breaking when we find the first unparsed element is accurate - for example:

```javascript
<amp-element>
  <amp-child1>
  <amp-child2>
  <amp-child3>
</amp-element>
```

Breaking at the first unparsed element will always break on the `amp-element` because it has not yet fully parsed although some of its children probably already did - so they're all delayed until its parent is fully parsed.